lib/reply.js

```diff
@@ -122,6 +122,7 @@ Reply.prototype.headers = function (headers) {
 
 Reply.prototype.code = function (code) {
   this.res.statusCode = code
+  this._headers['_hasStatusCode'] = true
```

Comment:
You can directly use `this._hasStatusCode`, without changing the headers object :)

Reply:
Okay, but how do I add the `_hasStatusCode` property to `this`? Or let me guess, it is initialized automatically?

Reply:
You must initialize it here: https://github.com/fastify/fastify/blob/36960965710756148cd86bc040d2c6f25ee4d04d/lib/reply.js#L24-L33 The default value should be `false`.